hedera-node/docs/design/services/smart-contract-service/update_token_custom_fees_sc.md

```diff
@@ -0,0 +1,87 @@
+
+## References
+
+[HIP-18](https://hips.hedera.com/hip/hip-18) - HIP that introduces token custom fees.\
+[HIP-206](https://hips.hedera.com/hip/hip-206) - HIP that enables the precompiled system contracts for Hedera Token Service.\
+[HIP-514](https://hips.hedera.com/hip/hip-514) - HIP that introduces token management via smart contracts.
+
+## Goals
+
+- Expose `updateFungibleTokenCustomFees` and `updateNonFungibleTokenCustomFees` as new functions in the Hedera Token Service Smart Contract.
+- Implement the needed HTS system contract classes to support the new function.
+
+## Non Goals
+
+- The implementation of the HAPI operation, as it is already an existing feature.
+
+## Architecture
+
+The architecture for update token custom fees follows the existing framework defined for handling all calls to the HederaTokenService system contract in the modularization services and is described in more detail in the Implementation section below.
+
+## Implementation
+
+### New Function Implementations
+
+New system contract functions must be added to the `IHederaTokenService` interface to support the updating the custom fees for fungible and non-fungible tokens.
+
+| Function Selector Hash |                                                                                        Function Signature                                                                                         |    Response    |                                 |
+|------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------------|---------------------------------|
+| `0xe780c5d3`           | `function updateFungibleTokenCustomFees(address token,  IHederaTokenService.FixedFee[] memory fixedFees, IHederaTokenService.FractionalFee[] memory fractionalFees) returns (int64 responseCode)` | `ResponseCode` | The response code from the call |
+| `0x01f9eb7d`           | `function updateNonFungibleTokenCustomFees(address token, IHederaTokenService.FixedFee[] memory fixedFees, IHederaTokenService.RoyaltyFee[] memory royaltyFees) returns (int64 responseCode)`     | `ResponseCode` | The response code from the call |
+
+### Smart Contract Service Module
+
+- UpdateTokenCustomFeesTranslator - This class will be responsible for handling the `updateFungibleTokenCustomFees` and `updateNonFungibleTokenCustomFees` selectors and dispatching them to the corresponding HAPI calls.
+- UpdateTokenCustomFeesDecoder - This class provides methods  and constants for decoding the given `HtsCallAttempt` into a `TransactionBody` for `TokenFeeScheduleUpdate` call.
```

Comment:
Does it make sense to add the specific TransactionBody that will be dispatched to the token service here for reference.

Reply:
Added the specific TransactionBody.